### ISSUE TITLE
[BUG] Default value for Checkbox and Radio Widgets

### ISSUE THREAD

**Issue by kookma** [TiddlyWiki/TiddlyWiki5#5930]
**Describe the bug**
The default value does not work for radio and checkbox widget if they bound to an index in a data tiddler.

**To Reproduce**
1. Go to https://tiddlywiki.com/prerelease
2. Create new tiddler with below text as body

```
<$checkbox tiddler=testTid field=ctest checked=yes default=yes> Checked</$checkbox>

<$radio tiddler=testTid field=rtest checked=yes default=yes> Selected</$radio>
```
3. The default value does not work,
4. Try to click on radio button it does not work

5. create a new tiddler with below text (here widget is bounded to a dataTiddler)

```
<$checkbox tiddler=testTid2 index=ctest checked=yes default=yes> Checked</$checkbox>

<$radio tiddler=testTid2  index=rtest checked=yes default=yes> Selected</$radio>
```

6. Select radio button, both are selected
7. See the error (both are selected)

**TiddlyWiki Configuration (please complete the following information):**
 - Version [e.g. v5.2.0]
 - Saving mechanism Timimi

 
**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser Chrome
 - Version 92

**Comment by pmario** [comment]
You are right there seems to be a bug in the prerelease

The radio widget has no  "checked" parameter.

**Comment by pmario** [comment]
Try this: 

```
<$radio tiddler=testTid field=rtest value=a default=a> Selected</$radio>
<$radio tiddler=testTid field=rtest value=b default=a> Selected</$radio>

<$radio tiddler=testTidX index=rtest value=a default=a> Selected</$radio>
<$radio tiddler=testTidX index=rtest value=b default=a> Selected</$radio>
```

**Comment by pmario** [comment]
There is some funny behaviour:

The following example works as expected, because "checked" and "unchecked" parameter are there

```
<$checkbox tiddler=testTid field=ctest checked=yes unchecked=no default=yes> Checked</$checkbox>
```

The following example DOESN'T work as expected, because "unchecked" parameter is MISSING

```
<$checkbox tiddler=testTid field=ctest checked=yes default=yes> Checked</$checkbox>
```

For the second example check and uncheck the widget. There's definitely something wrong

**Comment by kookma** [comment]
> The radio widget has no "checked" parameter. It has a "value" parameter

Yes, it just copy pasted the checkbox, and I forgot to correct the parameter for radio widget!
But it is still a bug!

**Comment by kookma** [comment]
Try this one in a new tiddler

```
<$checkbox tiddler=testTid2 index=ctest checked=yes unchecked=no default=yes> Checked</$checkbox>
```
Here the widget is bounded to a data tiddler! It behaves wrongly

**Comment by pmario** [comment]
> 
> 
> Try this one in a new tiddler
> 
> ```
> <$checkbox tiddler=testTid2 index=ctest checked=yes unchecked=no default=yes> Checked</$checkbox>
> ```
> 
> Here the widget is bounded to a data tiddler! It behaves wrongly

You are right, I can verify that's a problem

**Comment by pmario** [comment]
@Jermolene ... That seems to be a "blocking" bug for 5.2.0

**Comment by flibbles** [comment]
I don't know if I fixed everything, but PR #8964 should fix this problem where it concerns indexes, which was the only problem I encountered.
